tests/2_identify_generic_genes_pathways.ipynb

```diff
@@ -0,0 +1,740 @@
+{
```

Comment:
Following up on my comment in `human_analysis/2_identify_generic_genes_pathways.ipynb`, I guess I'm a bit confused on what the purpose of this test is. Are you trying to make sure this notebook runs without errors? If so, does that mean you'll need to remember to update these notebooks in the `tests` directory whenever you change the other notebooks?

I'm not necessarily opposed to this, just trying to get a handle on what your goal is with testing here. What you're doing is a bit different than the typical unit testing I see in other people's repos, so I just want to get your perspective on it.

Reply:
Thank you for thinking about this. My goal was to make sure that all the analysis runs without errors, so testing all the modules run at least. I tried to add assertion statements where I could (i.e. checking the VAE output and correlation outputs were the same). 

I have tried to add additional checks to test:
- that simulated experiments are reproducible
- that the ranks are reproducible

You're right that with the current setup I would need to update the test notebook if the analysis notebooks change to make sure they're up to date. Any advice you have to make these tests better is very welcome

Reply:
Makes sense. Maybe you could have a step in the workflow in `tests.yaml` to copy over the script from `human_analysis/2_identify_generic_genes_pathways.ipynb` to the `tests` directory? Then you wouldn't have the extra tests that you added just now, though, so you'd have to make a separate notebook to run those.

I don't think this needs to be done now - maybe you could create an issue for this and come back to it in the future?

Reply:
Thank you for thinking about this with me. So that analysis script is mainly calling other functions/modules and would be nice to automate but I'm ok manually porting things for now. I guess I'm not clear on how to write a script to port things over..